maven-modules/maven-plugins/jaxws/src/test/java/com/baeldung/soap/ws/client/CountryServiceClientUnitTest.java

```diff
@@ -0,0 +1,73 @@
+package com.baeldung.soap.ws.client;
+
+import org.junit.jupiter.api.*;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class CountryServiceClientUnitTest {
+
+    CountryServiceClient countryServiceClient;
+    CountryService countryService;
+
+    @BeforeEach
+    void setUp() {
+        countryService = mock(CountryService.class);
+        countryServiceClient = new CountryServiceClient(countryService);
+    }
+
+    @DisplayName("Get capital by country name when country not found")
+    @Test
+    void getCapitalByCountryName_WhenCountryDoesNotExists_ThenShouldThrowCountryNotFoundException() {
```

Comment:
Another interpretation of the BDD-style test naming convention, and perhaps a more popular one, is to use the givenX part for the precondition, and the whenY part for the method/action being invoked.

Using this approach, this test might be named: "givenCountryDoesNotExist_whenGetCapitalByCountryName_thenThrowsCountryNotFoundException"